taskcluster_util/util/downloader.py

```diff
@@ -54,23 +54,46 @@ def download_latest_artifact(self, task_id, full_filename, dest_dir):
         sys.stdout.flush()
 
         # move file to dest folder
-        abs_dest_dir = os.path.abspath(dest_dir)
-        log.debug('dest dir: {}'.format(abs_dest_dir))
+        abs_dest_dir = os.path.abspath(dest_dir) if dest_dir else os.getcwd()
```

Comment:
I think we need to check the path before doing the download. I just tried the edge case where the output dir is actually a file, I had to wait until the download is done before getting:
```
INFO: Starting new HTTPS connection (1): taskcluster-private-artifacts.s3-us-west-2.amazonaws.com
[##################################################] 276833131/276833131
Done.

WARNING: [/home/jlorenzo/git/taskcluster-util-python/lol] is not a folder.
INFO: Download [private/build/aries.zip] from TaskID [C4TOmBsMR_ucBc7JiB5aFg] to [/tmp/tmp_tcdl_GKMeaz/aries.zip] done.
```

I checked `/tmp/tmp_tcdl_GKMeaz/aries.zip`, this path doesn't exist.